pennychat/views.py

```diff
@@ -59,12 +74,19 @@ class UpdateDeleteFollowUp(mixins.UpdateModelMixin, mixins.DestroyModelMixin, ge
     """
     queryset = FollowUp.objects.all()
     serializer_class = FollowUpSerializer
+    permission_classes = [IsAuthenticated]
 
     def put(self, request, *args, **kwargs):
+        if self.get_object().user_profile.user != request.user:
```

Comment:
This could also get put into a mixin for this class  if every action is protected against unauthenticated users

Reply:
I actually meant to say that we could inherit from a base viewset that overrides get_object/ get_queryset to prevent users from accessing objects  that don't have access to.

Reply:
Ah! I like that idea.

Reply:
I decided to keep it as is. Since the rest framework generally uses permission classes, I'd like to stick to that paradigm, or use authentication decorators as necessary.